test/system/src/main/java/io/pravega/test/system/framework/DockerRemoteSequential.java

```diff
@@ -0,0 +1,171 @@
+        } catch (DockerException e) {
+            log.error("Unable to stop the test execution", e);
+        }
+    }
+
+    private CompletableFuture<Void> waitForJobCompletion() {
+        return Futures.loop(() -> isTestRunning(),
+                () -> Futures.delayedFuture(Duration.ofSeconds(3), executorService),
+                executorService);
+    }
+
+    private boolean isTestRunning() {
+        boolean value = false;
+        try {
+            if (Exceptions.handleInterrupted(() -> client.inspectContainer(this.id.get()).state().running())) {
+                value = true;
+            }
+        } catch (DockerException e) {
+            log.error("Unable to list docker services", e);
+        }
+        return value;
+    }
+
+    private String startTest(String containerName, String className, String methodName) {
+
+        try {
+            Exceptions.handleInterrupted(() -> client.pull(IMAGE));
+
+            ContainerCreation containerCreation = Exceptions.handleInterrupted(() -> client.createContainer(setContainerConfig(methodName, className), containerName));
+            assertFalse(containerCreation.id().toString().equals(null));
+
+            id.set(containerCreation.id());
+
+            final Path dockerDirectory = Paths.get(System.getProperty("user.dir"));
+            try {
+                //Copy the test.jar with all the dependencies into the container.
+                Exceptions.handleInterrupted(() -> client.copyToContainer(dockerDirectory, id.get(), "/data"));
+            } catch (Exception e) {
+                throw new TestFrameworkException(TestFrameworkException.Type.InternalError, "Unable to copy test jar to the container.Test failure", e);
+            }
+
+            String networkId = Exceptions.handleInterrupted(() -> client.listNetworks(DockerClient.ListNetworksParam.byNetworkName(DOCKER_NETWORK)).get(0).id());
+            //Container should be connect to the user-defined overlay network to communicate with all the services deployed.
+            Exceptions.handleInterrupted(() -> client.connectToNetwork(id.get(), networkId));
+
+            // Start container
+            Exceptions.handleInterrupted(() -> client.startContainer(id.get()));
+
+        } catch (DockerException e) {
+            log.error("Exception in starting container ", e);
+            Assert.fail("Unable to start the container to invoke the test.Test failure");
+        }
+        return id.get();
+    }
+
+    private ContainerConfig setContainerConfig(String methodName, String className) {
+
+        Map<String, String> labels = new HashMap<>(2);
+        labels.put("testMethodName", methodName);
+        labels.put("testClassName", className);
+
+        HostConfig hostConfig = HostConfig.builder()
+                .portBindings(ImmutableMap.of(DOCKER_CLIENT_PORT + "/tcp", Arrays.asList(PortBinding.of(LoginClient.MESOS_MASTER, DOCKER_CLIENT_PORT)))).networkMode("docker_gwbridge").build();
```

Comment:
Please break the long lines in this file, there are a few of them.

Reply:
formatted.